src/stores/toolbarStore.ts

```diff
@@ -1,41 +1,77 @@
 import { observable, computed, action } from 'mobx';
+import _ from 'lodash';
 import ToolbarTool from '~/enums/toolbarTool';
 import EditMode from '~/enums/editMode';
 
+import BaseTool from '~/tools/BaseTool';
+import AddNetworkNodeTool from '~/tools/AddNetworkNodeTool';
+import AddNewRoutePathTool from '~/tools/AddNewRoutePathTool';
+import CopyTool from '~/tools/CopyTool';
+import DivideLinkTool from '~/tools/DivideLinkTool';
+import EditNetworkNodeTool from '~/tools/EditNetworkNodeTool';
+import PrintTool from '~/tools/PrintTool';
+
+const TOOLS = [
+    new AddNetworkNodeTool(),
+    new AddNewRoutePathTool(),
+    new CopyTool(),
+    new DivideLinkTool(),
+    new EditNetworkNodeTool(),
+    new PrintTool(),
+];
+
+// TODO: Rename ToolbarStore -> toolStore?
 export class ToolbarStore {
-    @observable private _activeTool: ToolbarTool;
+    @observable private _selectedTool: BaseTool|null;
     @observable private _disabledTools: ToolbarTool[];
     @observable private _editMode: EditMode;
 
     constructor() {
         this._disabledTools = [
             ToolbarTool.Print,
         ];
-        this._activeTool = ToolbarTool.None;
     }
 
     @computed
-    get activeTool(): ToolbarTool {
-        return this._activeTool;
+    get selectedTool(): BaseTool | null {
+        return this._selectedTool;
     }
 
     @action setEditMode(editMode: EditMode) {
         this._editMode = editMode;
+        this.selectTool(null);
     }
 
-    @computed get editMode(): EditMode {
+    @computed
+    get editMode(): EditMode {
         return this._editMode;
     }
 
     @action
-    public toggleTool(tool: ToolbarTool) {
-        if (!this.isDisabled(tool)) {
-            this._activeTool = (this._activeTool === tool) ? ToolbarTool.None : tool;
+    public selectTool(tool: ToolbarTool | null) {
+        if (this._selectedTool) {
+            this._selectedTool.deactivate();
         }
+
+        // deselect current tool
+        if (tool === null || (this._selectedTool && this._selectedTool.toolType === tool)) {
+            this._selectedTool = null;
+            return;
+        }
+        const foundTool = _.find(TOOLS, (_tool) => {
```

Comment:
const foundTool = TOOLS.find(tool => tool.toolType === tool);
Do not use lodash if not necessary

Reply:
Having TOOLS as an object and selecting the tool by key would be even simpler.